### PR TITLE
Added custom calling context in answer api.

### PR DESCRIPTION
**Pull request by v-pivamshi**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallAutomationAsyncClient.java [hunk 1]
```diff
@@ -395,6 +397,13 @@ Mono<Response<AnswerCallResult>> answerCallWithResponseInternal(AnswerCallOption
                     = getTranscriptionConfigurationInternal(answerCallOptions.getTranscriptionConfiguration());
                 request.setTranscriptionConfiguration(transcriptionConfigurationInternal);
```
Comment (v-vdharmaraj) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallAutomationAsyncClient.java:398]: Please can you add the Unit tests and livetests

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAsyncClientUnitTests.java [hunk 2]
```diff
@@ -133,6 +133,29 @@ public void answerCall() {
...
+            Collections.singletonList(new AbstractMap.SimpleEntry<>(generateCallProperties(CALL_CONNECTION_ID,
+                CALL_SERVER_CALL_ID, CALL_CALLER_ID, CALL_CALLER_DISPLAY_NAME, CALL_TARGET_ID, CALL_CONNECTION_STATE,
+                CALL_SUBJECT, CALL_CALLBACK_URL, MEDIA_SUBSCRIPTION_ID, DATA_SUBSCRIPTION_ID), 200))));
+
+        AnswerCallOptions answerCallOptions = new AnswerCallOptions(CALL_INCOMING_CALL_CONTEXT, CALL_CALLBACK_URL)
+            .setMediaStreamingConfiguration(MEDIA_STREAMING_CONFIGURATION)
+            .setTranscriptionConfiguration(TRANSCRIPTION_CONFIGURATION);
+
+        Response<AnswerCallResult> answerCallResult
+            = callAutomationAsyncClient.answerCallWithResponse(answerCallOptions).block();
```
Comment (v-vdharmaraj) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAsyncClientUnitTests.java:148]: seems like missing the custom context on the unit test

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAsyncClientAutomatedLiveTests.java [hunk 3]
```diff
@@ -211,4 +212,121 @@ public void createVOIPCallAndRejectAutomatedTest(HttpClient httpClient) {
...
+
+         // wait for the incomingCallContext
+         String incomingCallContext = waitForIncomingCallContext(uniqueId, Duration.ofSeconds(10));
+         assertNotNull(incomingCallContext);
+
+         // answer the call
+         AnswerCallOptions answerCallOptions
+             = new AnswerCallOptions(incomingCallContext, DISPATCHER_CALLBACK + String.format("?q=%s", uniqueId));
+        if (answerCallOptions.getCustomCallingContext().getSipHeaders() != null
+            || answerCallOptions.getCustomCallingContext().getVoipHeaders() != null) {
```
Comment (v-vdharmaraj) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAsyncClientAutomatedLiveTests.java:278]: why do we need this?

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/AnswerCallOptions.java [hunk 4]
```diff
@@ -49,6 +56,7 @@ public final class AnswerCallOptions {
     public AnswerCallOptions(String incomingCallContext, String callbackUrl) {
         this.incomingCallContext = incomingCallContext;
         this.callbackUrl = callbackUrl;
```
Comment (v-vdharmaraj) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/AnswerCallOptions.java:58]: dont we needs to set this values in the constructor? does it work?

**Review by fangchen0601 — APPROVED** [review]

**Review by minwoolee-msft — APPROVED** [review]

**Review by cochi2 — APPROVED** [review]

**Pull request merged**